client/api/src/main/java/io/smallrye/graphql/client/core/utils/validation/NameValidation.java

```diff
@@ -0,0 +1,86 @@
+package io.smallrye.graphql.client.core.utils.validation;
+
+/**
+ * This class provides utility methods for validating names according to the GraphQL specification.
+ * 
+ * @see <a href="https://spec.graphql.org/draft/#Name">https://spec.graphql.org/draft/#Name</a>
+ */
+public class NameValidation {
+
+    /**
+     * The regular expression patterns for a valid GraphQL names.
+     */
+    private static final String NAME_PATTERN = "[a-zA-Z_][a-zA-Z0-9_]*";
```

Comment:
For performance purposes, can we compile these patterns into a `java.util.regex.Pattern` (using `Pattern.compile` and just once) and then use `pattern.matcher(actualString).matches()`? It will be faster than using `string.matches(pattern_as_string)` because that would mean recompiling the pattern every time again and again.